apps/web/src/hooks/useDocumentParameters/index.ts

```diff
@@ -418,65 +420,70 @@ export function useDocumentParameters<
     ],
   )
 
-  const onMetadataProcessed = useCallback(
+  const parameters = useMemo(
+    () => convertToParams(inputsBySource),
+    [inputsBySource],
+  )
+
+  const onMetadataChange = useCallback(
     (metadata: ConversationMetadata) => {
-      setInputs(
-        'manual',
-        recalculateInputs({
-          inputs: inputs.manual.inputs,
-          metadata,
-        }),
-      )
+      setValue((oldState) => {
+        const { doc } = getDocState(oldState, key)
+        console.log('MANUAL_INPUTS', doc.manual.inputs)
+        return updateInputsState({
```

Comment:
Here is where prev state is lost when someone comments a parameter in the prompt. Have to find a way of persisting prevState to recover and they restore that parameter